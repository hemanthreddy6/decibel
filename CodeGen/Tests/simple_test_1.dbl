print "hello world!";
f <- (a:long, b:int):int {
    if (a > b) {
        return a;
    } otherwise {
        return b;
    }
} ;
f1 <- (a:long, b:int):int {
    if (a < b) {
        return a;
    } otherwise {
        return b;
    }
} ;
print 1000hz;
f2 <- f;
if 100 > 20 {
    f2 = f1;
}
// If it's f1, 10 will be printed, otherwise 100
print f2(100, 10);
a <- 0;
loop 10 {
    a += 1;
    if a > 8 {
        print - a;
    } or a > 6 {
        print a * 100;
    } or a > 4 {
        print a - 100;
    } otherwise {
        print a;
    }
}

loop until a==0 {
    print a;
    a -= 1;
}

