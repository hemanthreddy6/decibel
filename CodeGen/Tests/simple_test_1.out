hello world!
14.213095
10
1
2
3
4
-95
-94
700
800
-9
-10
10
9
8
7
6
5
4
3
2
1
